### ISSUE TITLE
New Project Structure

### ISSUE THREAD

**Issue by SergueiFedorov** [ochagata/rateMyPaul#67]
Please let me know if this project structure makes sense:

https://github.com/SergueiFedorov/rateMyDePaul/tree/dev

The "BUILD_CHROME_PROJECT.py" script creates a "Chrome-Extension" automatically by bringing in all the scripts into a single package. You can then run the extension in the browser (refresh the browser plugins first, sorry!). All the work will be done in the Common/Chrome/Firefox/3rdParty folders and the script will compile them into a workable plugin. The same is done for the Firefox port.

Before commiting changes, you just run "GIT_CLEANUP_FOR_COMMIT.py" script which cleans up the "Chrome-Extension" and "Firefox-Plugin" folders so they don't get committed.

Basically we put the browser specific code in the FireFox/Chrome folders and shared code into the Common folder. Third Party stores things like jQuery.

Please let me know if this is too complicated!

**Comment by computerfr33k** [comment]
are these python 2 scripts? Just want to make sure if they are or not so I can add appropriate cross platform checking code to them. As I see the one is windows specific. (I run Linux on all my machines)

**Comment by SergueiFedorov** [comment]
The scripts are:

BUILD_CHROME_PROJECT.py - Builds the Chrome Project
GIT_CLEANUP_FOR_COMMIT.py - Clean up the Chrome project (will do for FireFox as well)
RUN_CHROME.py - Calls BUILD_CHROME_PROJECT.py and opens CampusConnect

I haven't added any cross platform code as of yet, so if you want to do that, that would be great! (why I went with python instead of batch)

**Comment by computerfr33k** [comment]
yea, I mean if you really wanted to be hardcore, you could go with c or c++. :)

**Comment by SergueiFedorov** [comment]
I don't care what C++ programmers say, C++ is an amazing language but I am not dealing with compiling C++ code on multiple platforms just to move files around LOL

**Comment by computerfr33k** [comment]
try these out, I hope they still work on windows.
https://gist.github.com/computerfr33k/921876416ba1eab589cb

**Comment by SergueiFedorov** [comment]
Works perfectly. One small change, it said "You chose not to continue" every time for the GIT_CLEANUP_FOR_COMMIT script. Small fix. Here is what I have:

https://github.com/SergueiFedorov/rateMyDePaul/commit/32a9e92c28624e89ae892acce9e779f3c06b8015

**Comment by SergueiFedorov** [comment]
New structure merged into Ozer's dev branch

**Comment by computerfr33k** [comment]
Something we could do so we don't need the GIT_COMMIT.py file, is to just put in a gitignore statement.
Would make it so you don't have to constantly delete the folders before commits. or require running a python script.
```
# .gitignore
.DS_Store
Chrome-Extension/
Firefox-Extension/
```

**Comment by SergueiFedorov** [comment]
This is a good idea!

**Comment by computerfr33k** [comment]
I can commit with my fix for updating my search_popup to use the new Get_Driver

**Comment by SergueiFedorov** [comment]
I think I ran through some of the code to convert it to use the Get_Driver stuff. I might have missed some (also, I am pretty sure I got everything in the rest of the code, but if you see anything that is using either $.get or Chrome.*, update it to the Get_Driver function calls
